--- tools/designer/uic/form.cpp.orig	2005-09-24 12:32:34.000000000 +0200
+++ tools/designer/uic/form.cpp	2005-09-24 12:33:28.000000000 +0200
@@ -726,15 +726,6 @@
 		}
 		n2 = n2.nextSibling().toElement();
 	    }
-        } else if ( n.tagName() == "includehints" ) {
-            QDomElement n2 = n.firstChild().toElement();
-            while ( !n2.isNull() ) {
-                if ( n2.tagName() == "includehint" ) {
-                    QString file = n2.firstChild().toText().data();
-                    localIncludes += file;
-                }
-                n2 = n2.nextSibling().toElement();
-            }
         }
     }
 
