--- Makefile.linux.orig	2003-10-05 17:07:29.000000000 +0200
+++ Makefile.linux	2003-10-05 17:11:29.000000000 +0200
@@ -29,6 +29,7 @@
 	'/usr/local/include/smpeg',
 	'/usr/include/SDL',
 	'/usr/include',
+        '/usr/include/SDL_console',
 	'/usr/include/smpeg',
 	'/usr/local/include/GL',
 	'/usr/local/include/gl',
@@ -48,7 +49,7 @@
 	SDL_net => { inc => 'HAVE_SDL_NET', test => 'SDL_net.h' },
 	SDL_ttf => { inc => 'HAVE_SDL_TTF', test => 'SDL_ttf.h' },
 	SDL_gfx => { inc => 'HAVE_SDL_GFX', test => 'SDL_gfxPrimitives.h' },
-	SDL_console => { inc => 'HAVE_SDL_CONSOLE', test => 'CON_console.h' },
+	SDL_console => { inc => 'HAVE_SDL_CONSOLE', test => 'SDL_console.h' },
 	png => { inc => 'HAVE_PNG', test => 'png.h' },
 	jpeg => { inc => 'HAVE_JPEG', test => 'jpeglib.h' },
 	GL => { inc => 'HAVE_GL', test => 'gl.h' },
