--- smb-method.c.orig	2005-09-04 13:47:18.000000000 +0200
+++ smb-method.c	2005-09-04 13:47:28.000000000 +0200
@@ -1425,7 +1425,7 @@
 
 		/* Important: perform_authentication leaves and re-enters the lock! */
 		while (perform_authentication (&actx) > 0) {
-			r = smb_context->close (smb_context, handle->file);
+			r = smb_context->close_fn (smb_context, handle->file);
 			actx.res = (r >= 0) ? GNOME_VFS_OK : gnome_vfs_result_from_errno ();
 		}
 
