--- Makefile.in.orig	2004-04-24 21:46:28.000000000 +0200
+++ Makefile.in	2004-04-24 21:46:33.000000000 +0200
@@ -1,6 +1,6 @@
 # $Id: Makefile.in,v 1.13.2.1 2003/07/28 08:23:09 VS Exp $
 
-CONTRIB_SUBDIRS=canvas fl gizmos mmedia net ogl plot stc svg xrc animate #applet
+CONTRIB_SUBDIRS=canvas fl gizmos mmedia net ogl plot stc svg xrc #applet
 
 all:
 	@for d in $(CONTRIB_SUBDIRS); do (cd $$d && $(MAKE)); done
