--- Makefile.orig	2004-01-07 16:45:24.000000000 +0100
+++ Makefile	2004-01-07 16:45:59.000000000 +0100
@@ -28,7 +28,7 @@
 .SUFFIXES:
 
 OBJS=apmlib.o
-EXES=apm apmd xapm apmsleep
+EXES=apm apmd apmsleep
 HEADERS=apm.h
 
 PREFIX=/usr
@@ -94,8 +94,8 @@
 	install        apmd            ${PREFIX}/sbin
 	install -b -Vt apmd_proxy      ${APMD_PROXY_DIR}
 	install -m 644 apmd.8          ${MANDIR}/man8
-	install        xapm            ${PREFIX}/bin
-	install -m 644 xapm.1          ${MANDIR}/man1/xapm.1
+	#install        xapm            ${PREFIX}/bin
+	#install -m 644 xapm.1          ${MANDIR}/man1/xapm.1
 	install        apmsleep	       ${PREFIX}/bin
 	install -m 644 apmsleep.1      ${MANDIR}/man1
 	install -m 644 apmsleep.1.fr   ${MANDIR}/fr/man1
